### PR TITLE
config-tools: refine MMIO32/64_START and MMIO32/64_END

### PR DESCRIPTION
**Pull request by yuchuyang**
Get the start and end address based on all pci bus memory resource.

Tracked-On: #6752
Signed-off-by: Yang,Yu-chu <yu-chu.yang@intel.com>

### CODE REVIEW

**Comment by yuchuyang** [comment]
waiting for https://github.com/projectacrn/acrn-hypervisor/pull/6747 to be merged in order to run CI.

**Comment by wenlingz** [comment]
@NanlinXie please help approve this PR, thanks

File: misc/config_tools/static_allocators/gpa.py [hunk 1]
```diff
@@ -297,8 +297,8 @@ def get_pt_devs_io_port_passthrough(board_etree, scenario_etree):
     return dev_list
 
 def get_pci_hole_native(board_etree):
-    resources = board_etree.xpath(f"//bus[@type = 'pci']/device[@address]/resource[@type = 'memory' and @len != '0x0']")
-    resources_hostbridge =  board_etree.xpath("//bus[@address = '0x0']/resource[@type = 'memory' and @len != '0x0' and not(starts-with(@id, 'bar')) and not(@width)]")
+    resources = board_etree.xpath(f"//bus[@type = 'pci']/device[@address]/resource[@type = 'memory']")
+    resources_hostbridge =  board_etree.xpath("//bus[@address = '0x0']/resource[@type = 'memory' and not(starts-with(@id, 'bar')) and not(@width)]")
```
Comment (junjiemao1) [inline misc/config_tools/static_allocators/gpa.py:301]: Why do we want to skip `@len != '0x0'` here? To get the resources assigned to all bridges/switches, it is the `@address = '0x0'` check that should be removed.

File: misc/config_tools/static_allocators/gpa.py [hunk 2]
```diff
@@ -361,10 +361,10 @@ def create_vuart_node(allocation_etree, vm_id, devdict):
 
 def create_native_pci_hole_node(allocation_etree, low_mem, high_mem):
     common.append_node("/acrn-config/hv/MMIO/MMIO32_START", hex(low_mem[0].start).upper(), allocation_etree)
-    common.append_node("/acrn-config/hv/MMIO/MMIO32_END", hex(low_mem[0].end + 1).upper(), allocation_etree)
+    common.append_node("/acrn-config/hv/MMIO/MMIO32_END", hex(low_mem[-1].end + 1).upper(), allocation_etree)
```
Comment (junjiemao1) [inline misc/config_tools/static_allocators/gpa.py:364]: So `low_mem` and `high_mem` are already sorted here?

**Review by junjiemao1 — CHANGES_REQUESTED** [review]

**Comment by wenlingz** [comment]
> waiting for #6747 to be merged in order to run CI.

6747 merged

File: misc/config_tools/static_allocators/gpa.py (hunk 1 above)
Comment (yuchuyang) [reply misc/config_tools/static_allocators/gpa.py]: resolved.

File: misc/config_tools/static_allocators/gpa.py (hunk 2 above)
Comment (yuchuyang) [reply misc/config_tools/static_allocators/gpa.py:365]: Yes, see the last time of get_pci_hole_native method.

>     return list(sorted(low_mem)), list(sorted(high_mem))

**Review by junjiemao1 — APPROVED** [review]

**Pull request merged**